src/evergreen/metrics/buildmetrics.py

```diff
@@ -193,12 +271,28 @@ def _count_task(self, task):
 
         if task.is_success():
             self.success_count += 1
+            if not task.generated_by:
+                self.display_success_count += 1
         else:
             self.failure_count += 1
+            if not task.generated_by:
+                self.display_failure_count += 1
             if task.is_system_failure():
                 self.system_failure_count += 1
+                if not task.generated_by:
+                    self.display_system_failure_count += 1
             if task.is_timeout():
                 self.timed_out_count += 1
+                if not task.generated_by:
+                    self.display_timed_out_count += 1
+
+        if task.generated_by:
+            if task.generated_by not in self._display_map:
```

Comment:
With the introduction of the enum, I think we can simplify this code.

First, I would change the declaration of `_display_map` to use a `defaultdict`:
```
from collections import defaultdict
...
self._display_map = defaultdict(list)
```
Then here, just append the task to `_display_map`:
```
if task.generated_by:
    self._display_map[task.generated_by].append(task)
```

Finally in `_count_display_tasks`, you can use the enum to determine the task status:
```
def _count_display_tasks(self):
    for generated_by, tasks in self._display_map.items():
        status = max([task.get_status_score() for task in tasks])
        if status == StatusScore.SUCCESS:
            self.display_success_count += 1
            continue
        if status == StatusScore.UNDISPATCHED:
            self.diplay_undispatched_count += 1
            continue
...
```